lib/semian/mysql2.rb

```diff
@@ -83,6 +83,20 @@ def query(*args)
       end
     end
 
+    # TODO: write_timeout and connect_timeout can't be configured currently
+    # dynamically, await https://github.com/brianmario/mysql2/pull/955
+    def with_resource_timeout(temp_timeout)
+      prev_read_timeout = @read_timeout
+      # C-ext reads this directly, writer method will configure
+      # properly on the client but based on my read--this is good enough
+      # until we get https://github.com/brianmario/mysql2/pull/955 in
+      @read_timeout = temp_timeout
+
+      yield
+    ensure
+      @read_timeout = prev_read_timeout
+    end
```

Comment:
In general this pattern makes me a bit uncomfortable when expressed like this. If someone were to add a line of code to the top of this method that might raise, the `ensure` could set a nil value. An alternative that more clearly maps to the intention is:

```
def
  prev = current
  begin
    current = temp
    # things
  ensure
    current = prev
  end
end
```